libs/velo-external-db-core/src/router.ts

```diff
@@ -225,12 +193,15 @@ export const createRouter = () => {
         }
     })
 
-    router.post('/data2/remove', async(req, res, next) => {
+    router.post('/data/remove', async(req, res, next) => {
         try {
             const removeRequest: dataSource.RemoveRequest = req.body;
             const collectionName = removeRequest.collectionId
-            
-            const objectsBeforeRemove = await removeRequest.itemIds.map(id => schemaAwareDataService.getById(collectionName, id))
+            const idEqExpression = removeRequest.itemIds.map(itemId => ({_id: {$eq: itemId}}))
+            const filter = {$or: idEqExpression}
+
+            const objectsBeforeRemove = (await schemaAwareDataService.find(collectionName, filterTransformer.transform(filter), undefined, 0, removeRequest.itemIds.length)).items
+
             const data = await schemaAwareDataService.bulkDelete(collectionName, removeRequest.itemIds)
             
```

Comment:
After discussing it with Max it seems like it is best to return it from the data provider itself. Since some databases like MongoDB can return the items they delete it is most efficient to use this ability. If we implement it in the outer layers we will have to do queries in order to find the items before their removal because we can't know if the data provider supports it or not. We will implement it this way in the future.